cmds/server.go

```diff
@@ -42,34 +45,47 @@ func newCmdServer() *cobra.Command {
 			storeProvider, err := store.NewStoreInterface(cfg)
 			term.ExitOnError(err)
 
-			err = runServer(storeProvider, conn)
+			err = runServer(storeProvider, conn, natsurl, logToNats)
 
 			//err = http.ListenAndServe(":4155", route(ctx, conn))
 			term.ExitOnError(err)
 			<-make(chan interface{})
 		},
 	}
+	_ = flag.CommandLine.Parse([]string{})
+
+	natsLogFlags := flag.NewFlagSet("nats-log-flags", flag.ExitOnError)
+	natslogr.InitFlags(natsLogFlags)
+
+	flag.VisitAll(func(f1 *flag.Flag) {
+		f2 := natsLogFlags.Lookup(f1.Name)
+		if f2 != nil {
+			_ = f2.Value.Set(f1.Value.String())
+		}
+	})
+
 	cmd.Flags().StringVar(&natsurl, "nats-url", "nats://localhost:4222", "Nats streaming server url")
 	cmd.Flags().StringVar(&clientid, "nats-client-id", "worker-p", "Nats streaming server client id")
+	cmd.Flags().BoolVar(&logToNats, "logToNats", false, "Publish logs to nats streaming server")
 
 	return cmd
 }
 
 //const ClientID = "worker-x"
 
-func runServer(storeProvider store.Interface, conn stan.Conn) error {
+func runServer(storeProvider store.Interface, conn stan.Conn, natsurl string, logToNats bool) error {
 
 	//defer apiserver.LogCloser(conn)
 
 	server := apiserver.New(conn)
-	err := server.CreateCluster(storeProvider)
+	err := server.CreateCluster(storeProvider, natsurl, logToNats)
```

Comment:
Can we just pass the logger here?

Reply:
No, we don’t know operation object here, that is used to create nats-logr, besides operation object will be different for every request